packages/flutter/test/material/dropdown_menu_test.dart

```diff
@@ -1110,6 +1110,79 @@ void main() {
     expect(textInput1.width, 200);
     expect(menu1.width, 200);
   });
+
+  testWidgets('Semantics does not include hint when input is not empty', (WidgetTester tester) async {
+    final ThemeData themeData = ThemeData();
+    const String hintText = 'I am hintText';
+    TestMenu? selectedValue;
+    final TextEditingController controller = TextEditingController();
+
+    await tester.pumpWidget(
+      StatefulBuilder(
+        builder: (BuildContext context, StateSetter setState) => MaterialApp(
+          theme: themeData,
+          home: Scaffold(
+            body: Center(
+              child: DropdownMenu<TestMenu>(
+                requestFocusOnTap: true,
+                dropdownMenuEntries: menuChildren,
+                hintText: hintText,
+                onSelected: (TestMenu? value) {
+                  setState(() {
+                    selectedValue = value;
+                  });
+                },
+                controller: controller,
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+    final SemanticsNode node = tester.getSemantics(find.text(hintText));
+
+    expect(selectedValue?.label, null);
+    expect(node.label, hintText);
+    expect(node.value, '');
+
+    await tester.tap(find.byType(DropdownMenu<TestMenu>));
+    await tester.pumpAndSettle();
+    await tester.tap(find.widgetWithText(MenuItemButton, 'Item 3').last);
+    await tester.pumpAndSettle();
+    expect(selectedValue?.label, 'Item 3');
+    expect(node.label, '');
+    expect(node.value, 'Item 3');
+  });
+
+  testWidgets('helperText is not visible when errorText is not null', (WidgetTester tester) async {
```

Comment:
Nice test! Can we also add two more tests to check if `DropdownMenu` can respect helperText and errorText when they are not null?

Reply:
Done 👍✅

Thanks a lot for the review!